artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/CoreAmqpConverter.java

```diff
@@ -530,4 +543,17 @@ private static byte destinationType(Destination destination) {
       throw new IllegalArgumentException("Unknown Destination Type passed to JMS Transformer.");
```

Comment:
I know it’s not your change but this looks like a nasty bug waiting to occur. Where a non jms client is being used

Reply:
There are only 4 types of JMS Queues.. all of them part of destinationType check

This is not used on JmsReply Any longer.

Reply:
The only bug that there is here, is JMS_DEST_TYPE_MSG_ANNOTATION, destinationType will always be a fixed Type, or Queue.

Reply:
I will deal with it on a separate PR:
https://issues.apache.org/jira/browse/ARTEMIS-2312
I marked as BLOCKER, so I won't release without fixing it.

Reply:
yes but what if NOT jms.....

Reply:
@michaelandrepearce there's currently an expectation that if you're sending AMQP to be converted, the message has to be in a certain format anyways. It's not always possible to convert everything between protocols.


If NOT JMS, send AMQP, receive AMQP, and don't deal with conversions.


This specific bit wouldn't be a problem though.


Also: non JMS Conversions are currently a problem regardless of what I'm doing here. it's beyond the scope.

Reply:
Fair enough to default it then, as long as we dont throw IllegalArgumentException i think that the main gotcha here.